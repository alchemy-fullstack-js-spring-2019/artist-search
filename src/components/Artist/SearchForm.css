@import url('https://fonts.googleapis.com/css?family=Titillium+Web&display=swap');

form {
  button{
    background-color: skyblue;
    border-radius: 5px;
    width: 8vw;
    height: 3vh;
    margin: 2vh;
  }

}
