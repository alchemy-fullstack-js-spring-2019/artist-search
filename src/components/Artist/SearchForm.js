import React from 'react';
import PropTypes from 'prop-types';
import styles from './SearchForm.css';


function SearchForm({ search, handleChange, searchSubmit }) {
  return (
    <form style={styles} onSubmit={ searchSubmit }>
      <input name="search" type="text" onChange={ handleChange } value={ search } /><br/>
      <button type="submit">Search</button>
    </form>
  );
}

SearchForm.propTypes = {
  search: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  searchSubmit: PropTypes.func.isRequired
};

export default SearchForm;
