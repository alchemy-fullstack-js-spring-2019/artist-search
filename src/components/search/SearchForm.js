import React from 'react';
import PropTypes from 'prop-types';

function SearchForm({ handleChange, handleSubmit, term }) {
  return (
    <form name="searchForm" onSubmit={handleSubmit} >
      <input type="text" name="searchTerm" onChange={handleChange} value={term}/>
      <button>Search</button>
    </form>
  );
}

SearchForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  term: PropTypes.string.isRequired
};

export default SearchForm;
