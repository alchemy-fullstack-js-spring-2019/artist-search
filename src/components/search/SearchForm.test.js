import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<SearchForm handleChange={jest.fn()} handleSubmit={jest.fn()} term='' />);
    expect(wrapper).toMatchSnapshot();
  });
});
