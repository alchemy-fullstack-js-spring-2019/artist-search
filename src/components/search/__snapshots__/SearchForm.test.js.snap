// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component matches snapshot 1`] = `
<form
  name="searchForm"
  onSubmit={[MockFunction]}
>
  <input
    name="searchTerm"
    onChange={[MockFunction]}
    type="text"
    value=""
  />
  <button>
    Search
  </button>
</form>
`;
